### PR TITLE
Adding firefox_installer.dart

### PR DESCRIPTION
**Pull request by nturgut**
It only downloads firefox right now. I'll add the installation in the next PR.

This PR also adds constants for firefox latest urls (linix, mac).

### CODE REVIEW

File: lib/web_ui/dev/common.dart [hunk 1]
```diff
@@ -54,6 +58,22 @@ class _LinuxBinding implements PlatformBinding {
...
+
+  @override
+  String getFirefoxDownloadUrl(String version) {
+    return 'https://download-installer.cdn.mozilla.net/pub/firefox/releases/${version}/linux-x86_64/en-US/firefox-${version}.tar.bz2';
+  }
+
+  @override
+  String getFirefoxExecutablePath(io.Directory versionDir) {
+    // TODO: implement getFirefoxExecutablePath
+    return null;
```
Comment (yjbanov) [inline lib/web_ui/dev/common.dart]: Maybe throw `UnimplementedError`?

File: lib/web_ui/dev/common.dart [hunk 2]
```diff
@@ -72,4 +92,34 @@ class _MacBinding implements PlatformBinding {
       'Contents',
       'MacOS',
       'Chromium');
+
+  @override
+  String getFirefoxDownloadUrl(String version) {
+    // TODO: implement getFirefoxDownloadUrl
+    return null;
```
Comment (yjbanov) [inline lib/web_ui/dev/common.dart]: same here

File: lib/web_ui/dev/common.dart [hunk 3]
```diff
@@ -72,4 +92,34 @@ class _MacBinding implements PlatformBinding {
...
+  }
+
+  @override
+  String getFirefoxExecutablePath(io.Directory versionDir) {
+    // TODO: implement getFirefoxExecutablePath
+    return null;
```
Comment (yjbanov) [inline lib/web_ui/dev/common.dart]: and here

File: lib/web_ui/dev/firefox_installer.dart [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  final RegExp forFirefoxVersion = RegExp("firefox-[0-9.]\+[0-9]");
+  final io.HttpClientRequest request = await io.HttpClient()
+      .getUrl(Uri.parse(PlatformBinding.instance.getFirefoxLatestVersionUrl()));
+  request.followRedirects = false;
+  // We will parse the HttpHeaders to find the redirect location.
+  final io.HttpClientResponse response = await request.close();
+
+  String version;
+
+  response.headers.forEach((String s, List<String> list) {
```
Comment (yjbanov) [inline lib/web_ui/dev/firefox_installer.dart]: I think you can do `response.headers.value('location')`, no? See: https://api.flutter.dev/flutter/dart-io/HttpHeaders/value.html

File: lib/web_ui/dev/common.dart (hunk 1 above)
Comment (nturgut) [reply lib/web_ui/dev/common.dart]: will be implemented soon. This is not a customer facing method.

File: lib/web_ui/dev/common.dart (hunk 2 above)
Comment (nturgut) [reply lib/web_ui/dev/common.dart]: same.

File: lib/web_ui/dev/common.dart (hunk 3 above)
Comment (nturgut) [reply lib/web_ui/dev/common.dart]: same.

File: lib/web_ui/dev/firefox_installer.dart (hunk 4 above)
Comment (nturgut) [reply lib/web_ui/dev/firefox_installer.dart]: Thanks! This is a lot nicer.

Comment (nturgut) [reply lib/web_ui/dev/firefox_installer.dart]: Done.

**Pull request merged**